### PR TITLE
Update location map generator, update locations with latest changes, update templates,update availability to include library name in front.

### LINKED ISSUE #1431: Reload locations and update templates before the first Alma production full index
The locations are updated in Alma but to be safe we should update the locations and the templates before we start the first Alma production full index. 
Reload locations both in bibdata-alma production and bibdata-alma staging

### LINKED ISSUE #1434: If holding library name is blank, location display should not fall back to the library name
This will update holdings1_display location field.
This will update the https://github.com/pulibrary/bibdata/blob/alma/marc_to_solr/translation_maps/holding_library.rb file.

improves https://github.com/pulibrary/orangelight/issues/2499

### LINKED ISSUE #1437: Update availability on the item and holding level to include the library in front of the location label
This is necessary so that the availability label that is called in orangelight through javascript to include the library label.
The location label comes from bibdata and is the value of the Alma external name.

### PR DESCRIPTION
**Pull request by christinach**
closes #1434 
closes #1431 
closes #1437
fixes the JS display for https://github.com/pulibrary/orangelight/issues/2499

### CODE REVIEW

File: app/services/location_maps_generator_service.rb [hunk 1]
```diff
@@ -21,7 +21,12 @@ def generate
     locations.each do |holding|
       holding_code = holding['code']
       lib_label = holding.library['label']
-      holding_label = holding['label'] == '' ? lib_label : holding['label']
+      # The locations.rb file needs to have the scsb locations as e.g "scsbnypl"=>"ReCAP".
+      # This is why we are pulling the library label only for the scsb locations.
+      # The rest of the holding_location_library records will take the holding['label'] value, even if it is blank.
+      # This change will reflect on the holdings1_display["location"] value since it is using the locations.rb file for the location field.
+      # see https://github.com/pulibrary/bibdata/blob/fe0a9b9495b7a8275f5a64463257c06537dd9c1e/marc_to_solr/lib/princeton_marc.rb#L653 for more info.
```
Comment (christinach) [inline app/services/location_maps_generator_service.rb:28]: This is something that I confirmed with @mzelesky.

File: spec/marc_to_solr/lib/princeton_marc_spec.rb [hunk 2]
```diff
@@ -664,6 +668,14 @@ def fixture_record(fixture_name)
       @holdings_scsb_block = @holdings_scsb[@holding_id_scsb]
     end
 
+    it 'indexes location if it exists' do
+      expect(@oversize_holding_block['location']).to eq 'Stacks'
+    end
+
+    it 'if location is blank it will index blank' do
+      expect(@holdings_no_location_name_holding_block["location"]).to eq ''
+    end
```
Comment (christinach) [inline spec/marc_to_solr/lib/princeton_marc_spec.rb:677]: This is a case mms_id : 99116547863506421, where the location code `eastasian$hy` has blank label (blank Alma external name) in bibdata. The `location` we index in holdings1_display is blank.

**Comment by christinach** [comment]
When you merge this PR you will need to update the locations on bibdata-alma-staging. 
- enter the rails console in the VM and run `LocationDataService.delete_existing_and_repopulate`
For local development you will need to generate from templates since the templates have been updated. This happens when you run `bundle exec rake servers:start`.

**Review by hackartisan — COMMENTED** [review]
I made some changes; someone else will need to review. Otherwise one inline comment about a new issue.

**Review by hectorcorrea — APPROVED** [review]

**Comment by hackartisan** [comment]
I ran `LocationDataService.delete_existing_and_repopulate`.

**Pull request merged**